### PR TITLE
Change sample table primary key

### PR DESCRIPTION
**Pull request by kpepper**
- Change sample table primary key to sample id instead of lane id
- Allow blank lane ids in sample uploads
- Fix logout bug
- Fix erroneous brief display of error dialog on sample upload screen

### CODE REVIEW

File: api/juno/api/models.py [hunk 1]
```diff
@@ -64,8 +64,8 @@ class HostStatus(models.TextChoices):
         MASTITIS = "mastitis"
         DISEASE_OTHER = "disease other"
 
-    lane_id = models.CharField(max_length=256, primary_key=True)
-    sample_id = models.CharField(max_length=256, unique=True)
+    sample_id = models.CharField(max_length=256, primary_key=True)
+    lane_id = models.CharField(max_length=256, unique=True, null=True)
```
Comment (puethe) [inline api/juno/api/models.py:68]: Do UNIQUE and nullable go well together? I.e. does this allow multiple entries that are NULL?

Comment (kpepper) [reply api/juno/api/models.py:68]: Yes, can have multiple nulls in a unique column, certainly for mysql and sqlite anyway.

**Review by garethpeat — APPROVED** [review]
@kpepper , looks good. Usage of `EMPTY_SAMPLE_RECORD_FIELD` suggests a null `laneId` must be passed over the API as the empty string, rather than `null`, so would suggest changing that (but maybe as a separate PR).

**Comment by kpepper** [comment]
**@garethpeat** Have added a Trello card for this.

**Pull request merged**